internal/provider/vdc_resource.go

```diff
@@ -0,0 +1,480 @@
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	sdkResource "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	apiclient "github.com/orange-cloudavenue/cloudavenue-sdk-go"
+
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/pkg/utils"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &vdcResource{}
+	_ resource.ResourceWithConfigure   = &vdcResource{}
+	_ resource.ResourceWithImportState = &vdcResource{}
+)
+
+// NewVdcResource is a helper function to simplify the provider implementation.
+func NewVdcResource() resource.Resource {
+	return &vdcResource{}
+}
+
+// vdcResource is the resource implementation.
+type vdcResource struct {
+	client *CloudAvenueClient
+}
+
+type vdcResourceModel struct {
+	Timeouts               timeouts.Value           `tfsdk:"timeouts"`
+	ID                     types.String             `tfsdk:"id"`
+	Name                   types.String             `tfsdk:"name"`
+	Description            types.String             `tfsdk:"description"`
+	VdcServiceClass        types.String             `tfsdk:"vdc_service_class"`
+	VdcDisponibilityClass  types.String             `tfsdk:"vdc_disponibility_class"`
+	VdcBillingModel        types.String             `tfsdk:"vdc_billing_model"`
+	VcpuInMhz2             types.Float64            `tfsdk:"vcpu_in_mhz2"`
+	CPUAllocated           types.Float64            `tfsdk:"cpu_allocated"`
+	MemoryAllocated        types.Float64            `tfsdk:"memory_allocated"`
+	VdcStorageBillingModel types.String             `tfsdk:"vdc_storage_billing_model"`
+	VdcStorageProfiles     []vdcStorageProfileModel `tfsdk:"vdc_storage_profiles"`
+	VdcGroup               types.String             `tfsdk:"vdc_group"`
+}
+
+type vdcStorageProfileModel struct {
+	Class   types.String `tfsdk:"class"`
+	Limit   types.Int64  `tfsdk:"limit"`
+	Default types.Bool   `tfsdk:"default"`
+}
+
+// Metadata returns the resource type name.
+func (r *vdcResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vdc"
+}
+
+// Schema defines the schema for the resource.
+func (r *vdcResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "VDC resource allows you to create a org VDC in Cloud Avenue.",
+		Attributes: map[string]schema.Attribute{
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+			}),
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The name of the org VDC. It must be unique in the organization. The length must be between 2 and 27 characters.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 27),
+				},
+			},
```

Comment:
If `Name` is the `ID`. Name [ForceNew](https://github.com/hashicorp/terraform-plugin-framework/blob/eebf3a50444518e7fe6ef210af70c32601bf3662/website/docs/plugin/framework/migrating/attributes-blocks/force-new.mdx )